process/interceptors/interceptedDataVerifier.go

```diff
@@ -39,38 +41,38 @@ func NewInterceptedDataVerifier(cache storage.Cacher) *interceptedDataVerifier {
 // It will retrieve the status in the cache if it exists, otherwise it will validate it and store the status of the
 // validation in the cache. Note that the entries are stored for a set period of time
 func (idv *interceptedDataVerifier) Verify(interceptedData process.InterceptedData) error {
+	hash := string(interceptedData.Hash())
+
 	if len(interceptedData.Hash()) == 0 {
 		return interceptedData.CheckValidity()
 	}
 
-	if val, ok := idv.cache.Get(interceptedData.Hash()); ok {
+	idv.km.RLock(hash)
```

Comment:
the critical section should keep the mutex as locked (not rlock) for the entire operation:
lock(hash)
defer unlock(hash)

cache.Get
check validity
cache.Put